cachi2/core/package_managers/yarn/project.py

```diff
@@ -35,12 +33,10 @@ class Plugin(TypedDict):
     spec: str
 
 
-class YarnRc:
-    """A yarnrc file.
+class ConfigurationFile(UserDict):
```

Comment:
Hmm, we modeled this as a module-scoped (private) ABC parent class in the yarn classic backend: https://github.com/containerbuildsystem/cachi2/blob/10144e70972af69fb20d9f682d576e0c799a5cfa/cachi2/core/package_managers/yarn_classic/project.py#L25, so I think ideally we'd want to be consistent with that, however, I suspect (haven't tried) you might face problems if you tried to inherit from both ABC and UserDict, especially if you tried to use ABC as a metaclass.
As much as I'd like to see us being consistent, it may not be the nicest path getting us there in this case. If anything, the class description should be changed or dropped.

Reply:
Oh damn, I completely missed that. I think there's no point in trying to use this as a base class (or mirror the approach), since the premise there is to have a foundation that allows manually mapping each necessary config option to an attribute, which is the approach we're trying to drop here for Yarn Berry.

I think we can either:
1- not be consistent, keep Yarn Berry's implemetation different and based on UserDict
2- try to use _CommonconfigFile as a base class and figure out a way to magically handle the attributes/options (Alexey's suggestion [here](https://github.com/containerbuildsystem/cachi2/pull/775#discussion_r1901368985))
3- Drop this PR altogether (we probably won't need to add many more config options in the near future)

Reply:
Yeah, I think we could even converge to using a simple UserDict in both Yarn Berry and Classic. I agree that the current PR already brings us benefits the way it is.

Reply:
Agreed with 1. For the record I wasn't trying to advocate for porting the ABC stuff from classic here as I find that approach inferior to this one.